tree/ntuple/v7/inc/ROOT/libdaos_mock/daos.h

```diff
@@ -201,33 +212,27 @@ enum {
 	DAOS_REC_ANY		= 0,
 };
 
-static inline int daos_obj_generate_id(daos_obj_id_t *oid, daos_ofeat_t ofeats,
-		     daos_oclass_id_t cid, uint32_t args)
-{
-	(void)args;
-	uint64_t hdr;
-
-	/* TODO: add check at here, it should return error if user specified
-	 * bits reserved by DAOS
-	 */
-	oid->hi &= (1ULL << OID_FMT_INTR_BITS) - 1;
-	/**
-	 * | Upper bits contain
-	 * | OID_FMT_VER_BITS (version)		 |
-	 * | OID_FMT_FEAT_BITS (object features) |
-	 * | OID_FMT_CLASS_BITS (object class)	 |
-	 * | 96-bit for upper layer ...		 |
-	 */
-	hdr  = ((uint64_t)OID_FMT_VER << OID_FMT_VER_SHIFT);
-	hdr |= ((uint64_t)ofeats << OID_FMT_FEAT_SHIFT);
-	hdr |= ((uint64_t)cid << OID_FMT_CLASS_SHIFT);
-	oid->hi |= hdr;
-
-	return 0;
-}
+#define DAOS_OCH_RDD_BITS 4
+#define DAOS_OCH_SHD_BITS 6
+#define DAOS_OCH_RDD_SHIFT 0
+#define DAOS_OCH_SHD_SHIFT DAOS_OCH_RDD_BITS
+#define DAOS_OCH_RDD_MAX_VAL ((1ULL << DAOS_OCH_RDD_BITS) - 1)
+#define DAOS_OCH_SHD_MAX_VAL ((1ULL << DAOS_OCH_SHD_BITS) - 1)
+#define DAOS_OCH_RDD_MASK (DAOS_OCH_RDD_MAX_VAL << DAOS_OCH_RDD_SHIFT)
+#define DAOS_OCH_SHD_MASK (DAOS_OCH_SHD_MAX_VAL << DAOS_OCH_SHD_SHIFT)
+
+/// Flags for oclass hints
+enum {
+   /// Flags to control OC Redundancy
+   DAOS_OCH_RDD_DEF = (1 << 0), // Default - use RF prop
+   DAOS_OCH_RDD_NO = (1 << 1),  // No redundancy
+   DAOS_OCH_RDD_RP = (1 << 2),  // Replication
+   DAOS_OCH_RDD_EC = (1 << 3),  // Erasure Code
+};
```

Comment:
We may need to add `DAOS_OCH_SHD_DEF`, if eventually used in RDaos.cxx.  OTOH, some of the other enumerators can be removed (unused); the less lines we have to maintain here, the better.